src/Data/Api.re

```diff
@@ -0,0 +1,175 @@
+let faunaUrl = "https://graphql.fauna.com/graphql";
+
+let fetchConfessionsRequestInit = {
+  let allConfessionsQueryPayload = Js.Dict.empty();
+
+  Js.Dict.set(
+    allConfessionsQueryPayload,
+    "query",
+    Js.Json.string(Queries.allConfessionsQuery),
+  );
+
+  Fetch.RequestInit.make(
+    ~method_=Post,
+    ~body=
+      Fetch.BodyInit.make(
+        Js.Json.stringify(Js.Json.object_(allConfessionsQueryPayload)),
+      ),
+    ~headers=
+      Fetch.HeadersInit.make({"Authorization": "Bearer " ++ Env.token}),
+    (),
+  );
+};
+
+let fetchConfessions = callback => {
+  Js.Promise.(
+    Fetch.fetchWithInit(faunaUrl, fetchConfessionsRequestInit)
+    |> then_(Fetch.Response.json)
+    |> then_(json =>
+         json
+         |> ConfessionData.Decode.decodeGetRecentConfessionsResponse
+         |> (
+           decodedResponse =>
+             callback(
+               Some(decodedResponse.data.allConfessions.allConfessionsArray),
+             )
+             |> resolve
+         )
+       )
+    |> catch(_error => callback(None) |> resolve)
```

Comment:
blindly catching and doing nothing here is a big code smell; you just don't know what the dev (even if that's you) is going to want to use this for, so unless you have a good reason, you generally want to forward the errors back to the invoker, and they can decide what to do.

ignoring errors here COULD be a violation of the single property principle from SOLID, since this function both wraps the API call AND handles error (but that might be a bit of a stretch)